src/main/java/org/maxgamer/quickshop/util/Util.java

```diff
@@ -1330,4 +1331,19 @@ public static String createRegexFromGlob(@NotNull String glob) {
         return out.toString();
     }
 
+    /**
+     * Get location that converted to block position (.0)
+     *
+     * @param loc location
+     * @return blocked location
+     */
+    @NotNull
+    public static Location getBlockLocation(@NotNull Location loc) {
+        loc = loc.clone();
+        loc.setX(loc.getBlockX());
+        loc.setY(loc.getBlockY());
+        loc.setZ(loc.getBlockZ());
+        return loc;
```

Comment:
What about  just ```return new Location(loc.getWorld(),loc.getBlockX(),loc.getBlockY(),loc.getBlockZ());``` ?

Reply:
may lose pitch and yaw

Reply:
Just a idea, if you want to use it, just add pitch and yaw in there aswell